datumaro/components/media.py

```diff
@@ -741,44 +983,163 @@ class RoIImage(Image):
 
     def __init__(
```

Comment:
I think that `ROIImage` and `MosaicImage` are secondary media types which can only be created from the first (e.g. `Image`), not by itself. Therefore, it would be good to keep an only one constructor (`cls.create_from_image()`).

Reply:
https://github.com/openvinotoolkit/datumaro/pull/921/commits/ef3b3b8dcc62baede633ce700a29f1c3fa89169b